p4-16/spec/P4-16-spec.mdk

```diff
@@ -3610,6 +3610,29 @@ expressions of type `int` must be compile-time known values.  The type
   The right operand must be either an unsigned constant of type `bit<S>` or a non-negative integer compile-time known value.
   The expression `a << b` is equal to $a \times 2^b$ while `a >> b`
   is equal to $\lfloor{a / 2^b}\rfloor$.
+- Extraction of a set of contiguous bits, also known as a slice, denoted by `[m:l]`,
+  where `m` and `l` must be expressions that evaluate to
+  non-negative compile-time known values, and `m >= l`. The types of `m` and `l` (which do not have to be identical)
+  must be one of the following:
+
+  - `int` - an infinite-precision integer (section
+    [#sec-infinite-precision-integers])
+  - `bit<W>` - a `W`-bit unsigned integer where `W >= 0` (section
+    [#sec-unsigned-integers])
+  - `int<W>` - a `W`-bit signed integer where `W >= 1` (section
+    [#sec-signed-integers])
+  - a serializable `enum` with an underlying type that is `bit<W>` or
+    `int<W>` (section [#sec-enum-types]).
+
+  The result is an unsigned bit-string of width `m - l + 1`, including the bits numbered
+  from `l` (which becomes the least significant bit of the result) to `m` (the
+  most significant bit of the result) from the source operand.
```

Comment:
Technically for int types we don't mandate a representation, so it is not clear what the bits of the source operand are.

Reply:
You will probably hear this multiple times from multiple sources, but Section 7.1 of the spec says "Fixed-width signed integers represented using two's complement int<>"

Reply:
I agree that doesn't specify for `int` types, only `int<W>`, but I'm all in favor of specifying it as two's complement for `int` as well, if that helps move the ball along.

Reply:
Yes, that's my point. We don't mandate such a representation for int, but when we apply a slice we compute the int representation as a m-wide number using 2's complement. I think that should be part of the text.